lib/staged_event/publisher/google_pub_sub.rb

```
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require "google/cloud/pubsub"
+
+module StagedEvent
+  module Publisher
+    class GooglePubSub < Base
+      include ShortCircuIt
+
+      def publish(model)
+        topic_id = topic_map.fetch(model.topic, topic_map.values.first)
```

this seems error-prone. nothing is checking the presence of `topic_map` for having at least one value, and also it silently uses a different topic. I think the behavior should be simplified, and custom errors created for:

* the map is empty
* the map does not contain the desired topic

---

Ah regarding errors, yeah hadn't quite finished this - at the very least it needs to translate "expected" exceptions from Google into `Publisher::PublishFailedError`, so the publisher process rescues them.

But re: the topic map, in the meantime I was relying on someone running `rails g staged_event::install` and getting the initializer added to their project, which initializes the topic map. Agree it should be more robust 👍